mmdet/models/dense_heads/nasfcos_head.py

```diff
@@ -0,0 +1,81 @@
+import copy
+
+import torch.nn as nn
+from mmcv.cnn import ConvModule, Scale, bias_init_with_prob, normal_init
+
+from mmdet.models.dense_heads.fcos_head import FCOSHead
+from ..builder import HEADS
+
+
+@HEADS.register_module()
+class NASFCOSHead(FCOSHead):
+    """Anchor-free head used in NASFCOS <https://arxiv.org/abs/1906.04423>
+
+    It is quite similar with FCOS head, except for the searched structure
+    of classification branch and bbox regression branch, where a structure
+    of "dconv3x3, conv3x3, dconv3x3, conv1x1"  is utilized instead.
+
+     Args:
+        num_classes (int): Number of categories excluding the background
+            category.
+        in_channels (int): Number of channels in the input feature map.
+        arch (list): A list composed of dicts, where each dict defines
+            a convolution layer correspondingly, includes type, kernel,
+            padding, etc.
+
+    Example:
+        >>> import torch
+        >>> arch = [dict(type='Conv', kernel_size=1)]
+        >>> self = NASFCOSHead(11, 7, arch)
+        >>> feats = [torch.rand(1, 7, s, s) for s in [4, 8, 16, 32, 64]]
+        >>> cls_score, bbox_pred, centerness = self.forward(feats)
+        >>> assert len(cls_score) == len(self.scales)
+    """
+
+    def __init__(self, num_classes, in_channels, arch, **kwargs):
```

Comment:
Is it necessary to make the arch configurable? NASFCOSHead indicates the specific arch of "dconv3x3, conv3x3, dconv3x3, conv1x1" in my view.